### ISSUE TITLE
Project Management

### ISSUE THREAD

**Issue by vedez** [vedez/aga_documentation#5]
- [x] Meeting Logs
Create a document/diary for this. Also add a reminder to do this weekly. 
Add everything you do on the issue board to keep track of what you have done for this project.

- [x] Project Plan
Divided the project into 4 Quarters, 2Qs each semester. 1st semester's focused on development and 2nd semester is focued on refinement.

- [x] Milestones
- > Add this into a table/timeline for documentation (include dates).
- Feasibility Research
- Product Design
- Version 00: Prototype & Review
- Vesion 01: Code Implementation (Base) with Testing & Review
- Version 02: Code Refinement with Testing & Review
- Final Documentation 
- Launch Release / Deployment**
- Official Website (One-Page)
- Project Completion

- [x] Roadmap
**Once Issues are added, roadmap can be overviewed on GB automatically.**

- [x] Task Distribution
- To Do: Stories, Tasks, Bugs, etc. (Labels) All possible Issues for this project. 
- > Each iteration will have specific Issues to be completed and if incompleted, it stays in backlog until allocated time is dedicated for this to catch-up.
- In Progress: One Task at a time. This organises what is being completed from what is left to do. It will also avoid feeling overwhelmed as the focus is only on one task at a time. Stories are to be divided.
- Review &  Testing: Once a feature is completed, this can be comitted into a branch. Once reviewed and meets standards, the branch is merged into the main project. This reduces possible bugs/errors.
- Done: A task is completed once it meets all requirements, and standard level.
- Priority and weight labels: Highlights what tasks are to be done first. (0 - 2 Levels)

- [x] Supervisor Engagement
- Task: 
- > Project Proposal & Summary
- > Explain Product and Project Management, Milstones/Roadmap (GB - Plan Breakdown), Engagement Expectations.
- Meeting Availability 
- > Tuesday 14:00 - 15:00
- > Wednesday 11:00 - 15:00
- > Thursday 11:00 - 15:00

- Time Capped (Max)
- > 15 min. Catch-up (Call or Email (PDF of what has been completed with explaination.))
- > 30 min. Review (Features, product, testing, etc.) (Call with Demo)
- > 60 min. Full Review & Retro per Iteration (Demo & Improving communication with supervisor)

- [x] Risk Management
- Technical Risks
- > Software Bugs and Errors: How to mitigate this? Regular Testing? Review Issue Board? Approach to non-functioning features?
- > Usage of branches to have different versions and only merge when reviewed.
- > Security/Authorisation level? OOP approach?

- > User Risks (Satisfactory Level?)
- UX Failures: UX Testing Cases, prioritise user feedback and improve usability 
- Communication with users is key as this is they are directly using the application for their benefits. Understand what they need over what they want.

- > Project Management
- Time Constraints? Estimate how long each task/story needs. Plan for unexpected delays - buffer, netfall?
- Scope Creep: Define the project's scope (main core features, etc.) and any additions to backlog (to do if there's time). Avoid derailing from the plan or over-spending hours.
- Freeze point needed.

- > Legal and Regulatory Risks
- Data Privacy Laws Compliance (GDPR)
- Age Restriction to devices. How safe is the product for younger audiences?
- Other laws to consider?

- > ChatGPT Notes:
Steps for Effective Risk Management:
- Risk Identification: List all potential risks from the above categories.
- Risk Assessment: Assess the likelihood of each risk occurring and the impact it could have on your project.
- Risk Prioritization: Focus on the highest-priority risks (those with both high likelihood and high impact).
- Mitigation Planning: Develop strategies to mitigate each identified risk. For example, have contingency plans, backup resources, or additional testing phases.
- Monitoring: Continuously monitor for new risks and the status of identified risks as the project progresses.
